### ISSUE TITLE
obs lua signal_handler_disconnect not working

### ISSUE THREAD

**Issue by Janzert** [obsproject/obs-studio#3841]
tldr: #3218 also exists for the lua implementation. The fix is simply changing the `!=` to `==` in [obs-scripting-lua.c:line 459](https://github.com/obsproject/obs-studio/blob/cab32e7e67c2a78d10e0961f746ee7408978fb43/deps/obs-scripting/obs-scripting-lua.c#L459)

## Platform
Operating system and version: All platforms
OBS Studio version: 21.0 - 26.2rc2

## Expected Behavior
Calling `signal_handler_disconnect` in lua should remove the signal handler callback.

## Current Behavior
The callback is either not removed or if it's registered for multiple signals on one signal handler the wrong signal is disconnected.

## Steps to Reproduce
1. Save and load the following script:
```lua
obs = obslua

function h1(cb)
    print("handler 1 called")
end

function h2(cb)
    print("handler 2 called")
end

function script_load()
    local obs_sh = obs.obs_get_signal_handler()
    obs.signal_handler_connect(obs_sh, "source_volume", h1)
    obs.signal_handler_connect(obs_sh, "source_activate", h1)
    obs.signal_handler_connect(obs_sh, "source_activate", h2)
    obs.signal_handler_disconnect(obs_sh, "source_activate", h1)
    print("h1 should be registered with volume changes and h2 with activation")
end
```
2. Open the script log
3. Try activating a source and changing a volume level
4. See that handler 1 and handler 2 both trigger on source activation, instead of handler 1 on volume changes and handler 2 on activation
